### PR TITLE
feat: Implement StepProperties tab

### PR DESCRIPTION
**Pull request by zoltanszabo-bitrise**
CI-2554

### CODE REVIEW

File: source/javascripts/_componentRegister.js [hunk 1]
```diff
@@ -116,5 +113,14 @@ angular
 	.component("rNavigation", register(Navigation, ["items", "activeItem", "onItemSelected"]))
 	.component(
 		"rStepConfig",
-		register(StepConfig, ["step", "highlightVersionUpdate", "outputVariables", "onClone", "onRemove"]),
+		register(StepConfig, [
+			"workflowIndex",
```
Comment (AndrasEszes) [inline source/javascripts/_componentRegister.js:117]: please...

File: source/javascripts/components/StepConfig.tsx [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
 
-const StepConfig = ({ step, highlightVersionUpdate, outputVariables, onClone, onRemove }: Props): JSX.Element => {
+const StepConfig = ({
+	step,
+	onClone,
+	onRemove,
+	hasVersionUpdate,
+	versionsWithRemarks,
+	outputVariables,
+	onPropertyChange,
```
Comment (AndrasEszes) [inline source/javascripts/components/StepConfig.tsx:27]:
```suggestion
	onChange,
```

File: source/javascripts/components/StepProperties.tsx [hunk 3]
```diff
@@ -1,17 +1,205 @@
...
+			oldVersion: step.defaultStepConfig.version,
+			oldInputNames: extractInputNames(step),
+		});
+		onChange(values);
+	});
+
+	useEffect(() => {
+		setValue("name", name);
+		setValue("version", version);
+	}, [name, version, setValue]);
```
Comment (AndrasEszes) [inline source/javascripts/components/StepProperties.tsx:125]: Investigate why is it necessary here. Maybe the useForm stores the default values and it doesn't re-initialize on step changes.

File: source/javascripts/controllers/_MainController.js.erb [hunk 4]
```diff
@@ -655,6 +655,7 @@ import datadogRumCustomTiming from "../utils/datadogCustomRumTiming.ts";
 				}
 
 				$rootScope.$emit("MainController::changesDiscarded");
+				$scope.$digest();
```
Comment (AndrasEszes) [inline source/javascripts/controllers/_MainController.js.erb:658]:
```suggestion
				safeDigest($scope);
```

File: source/javascripts/controllers/_WorkflowsController.js.erb [hunk 5]
```diff
@@ -1,1350 +1,1435 @@
...
 
-				viewModel.stackForWorkflow = function(workflow, shouldNormalize) {
-					if (shouldNormalize === undefined) {
-						shouldNormalize = false;
-					}
+			viewModel.selectedStep;
+			viewModel.selectedStepVersion;
+			viewModel.selectedStepLatestVersion;
+			viewModel.isSelectedStepLatestVersion;
+			viewModel.selectedStepVersionsWithRemarks;
```
Comment (AndrasEszes) [inline source/javascripts/controllers/_WorkflowsController.js.erb:776]: Please review usage

File: source/javascripts/controllers/_WorkflowsController.js.erb [hunk 6]
```diff
@@ -1,1350 +1,1435 @@
...
+				changeStepToVersion(viewModel.selectedStep, version);
+			};
 
-				$scope.$watch(
-					function() {
-						return viewModel.selectedStep ? viewModel.selectedStep.requestedVersion() : undefined;
-					},
-					function(selectedStepRequestedVersion) {
-						viewModel.requestedStepVersion = selectedStepRequestedVersion;
+			viewModel.upgradeStepToLatestVersion = function(step, workflowIndex) {
```
Comment (AndrasEszes) [inline source/javascripts/controllers/_WorkflowsController.js.erb:1012]: Please review usage

File: source/javascripts/controllers/_WorkflowsController.js.erb [hunk 7]
```diff
@@ -1,1350 +1,1435 @@
...
-						if (newRenamedStep == null) {
-							if (Step.isValidTitle(viewModel.renamedStepTitle)) {
-								oldRenamedStep.title(viewModel.renamedStepTitle);
-							}
+				if (Step.isValidTitle(newTitle)) {
+					viewModel.renamedStepTitle = newTitle;
+					viewModel.selectedStep.title(newTitle);
+					safeDigest($scope);
+				}
+			};
```
Comment (AndrasEszes) [inline source/javascripts/controllers/_WorkflowsController.js.erb:1120]: Please review usage

**Review by AndrasEszes — CHANGES_REQUESTED** [review]

File: source/javascripts/components/StepProperties.tsx (hunk 3 above)
Comment (zoltanszabo-bitrise) [reply source/javascripts/components/StepProperties.tsx:125]: Unfortunately, that's the only way I could get it work.

**Review by AndrasEszes — APPROVED** [review]

**Review by AndrasEszes — APPROVED** [review]

**Pull request merged**